crates/core/src/client/errors.rs

```diff
@@ -35,10 +34,10 @@ pub enum ConfigError {
 
 /// Error that can accure when interacting with the Kakarot ETH API.
 #[derive(Debug, Error)]
-pub enum EthApiError<T: std::error::Error> {
+pub enum EthApiError<E: std::error::Error> {
     /// Request to the Starknet provider failed.
     #[error(transparent)]
-    RequestError(#[from] ProviderError<JsonRpcClientError<T>>),
+    RequestError(#[from] ProviderError<E>),
```

Comment:
Here we had JsonRpcClientError<T> before, when using JsonRpc and getting and error, would this format work?

Reply:
We needed something more modular